File: WebClient SDK/src/PnrScripts/UI/PnrPowerAttorneyUIExtension.ts
import { CardUIExtension, ICardUIExtensionContext, IControlViewModel } from 'tessa/ui/cards';
import { Guid, Visibility } from 'tessa/platform';
import { PnrCardTypes } from '../Shared/PnrCardTypes';

export class PnrPowerAttorneyUIExtension extends CardUIExtension {
  public initialized(context: ICardUIExtensionContext) {
    if (!Guid.equals(context.card.typeId, PnrCardTypes.PnrPowerAttorneyTypeID)) {
      return;
    }
    let PnrPowerAttorney = context.card.sections.tryGet("PnrPowerAttorney");
    let confidants = context.model.controls.get('Confidants'); // Доверенное лицо
    let confidantNotEmployee = context.model.controls.get('ConfidantNotEmployee'); // Доверенное лицо (не сотрудник компании)
    if (!PnrPowerAttorney) return;
    
    if(confidants != null && confidantNotEmployee != null)
    {
      // добавляем валидацию контролы 'Доверенное лицо' и 'Доверенное лицо (не сотрудник компании)'
      //confidants.validationFunc = control => control != null && control.hasEmptyValue ? "Поле 'Доверенное лицо' обязательно для заполнения!" : null;
      //confidantNotEmployee.validationFunc = control => control != null && control.hasEmptyValue ? "Доверенное лицо (не сотрудник компании)' обязательно для заполнения!" : null;
      // видимость контролов при открытии карточки на редактирование
      if (PnrPowerAttorney.fields.get('EmployeeID') != null) {
        this.SetConfidantsVisibility(PnrPowerAttorney.fields.get('EmployeeID'), confidants, confidantNotEmployee);
      }
      // видимость контролов при смене значения "Сотрудник ГК Пионер"
      PnrPowerAttorney.fields.fieldChanged.add((e) => {
        if (e.fieldName == "EmployeeID" && e.fieldValue != null && confidants && confidantNotEmployee)
        {
          this.SetConfidantsVisibility(e.fieldValue, confidants, confidantNotEmployee);
        }
      });
    }
  }

  /** установка видимости контролов Доверенное лицо. 
   * employee: 1 - сотрудник, 0 - нет
  */
  private SetConfidantsVisibility(employeeID, confidants: IControlViewModel, confidantNotEmployee: IControlViewModel)
  {
    confidants.controlVisibility = employeeID == 1 ? Visibility.Visible : Visibility.Collapsed;
    confidantNotEmployee.controlVisibility = employeeID == 0 ? Visibility.Visible : Visibility.Collapsed;
  }
}

File: Extensions/Tessa.Extensions.Client/UI/PnrPowerAttorneyUIExtension.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using Tessa.Cards;
using Tessa.UI.Cards;
using Tessa.UI.Cards.Controls;

namespace Tessa.Extensions.Client.UI
{
    public sealed class PnrPowerAttorneyUIExtension : CardUIExtension
    {
        // установка видимости контролов Доверенное лицо
        // employee: 1 - сотрудник, 0 - нет
        private void SetConfidantsVisibility(int employeeID, IControlViewModel confidants, IControlViewModel confidantNotEmployee)
        {
            confidants.ControlVisibility = employeeID == 1 ? Visibility.Visible : Visibility.Collapsed;
            confidantNotEmployee.ControlVisibility = employeeID == 0 ? Visibility.Visible : Visibility.Collapsed;
            confidants.Block.Rearrange();
        }
        public override Task Initialized(ICardUIExtensionContext context)
        {
            Card card = context.Card;
            ICardModel cardModel = context.Model;

            IControlViewModel confidants;                // Доверенное лицо
            IControlViewModel confidantNotEmployee;     // Доверенное лицо (не сотрудник компании)
            cardModel.Controls.TryGet("Confidants", out confidants);
            cardModel.Controls.TryGet("ConfidantNotEmployee", out confidantNotEmployee);
            
            if(confidants != null && confidantNotEmployee != null)
            {
                // добавляем валидацию контролы 'Доверенное лицо' и 'Доверенное лицо (не сотрудник компании)'
                //confidants.ValidationFunc = c =>
                //    (((ControlViewModelBase)c).HasEmptyValue())
                //        ? $"Поле 'Доверенное лицо' обязательно для заполнения!"
                //        : null;
                //confidantNotEmployee.ValidationFunc = c =>
                //    (((ControlViewModelBase)c).HasEmptyValue())
                //        ? $"'Доверенное лицо (не сотрудник компании)' обязательно для заполнения!"
                //        : null;

                // видимость контролов при открытии карточки на редактирование
                if (card.Sections["PnrPowerAttorney"].Fields["EmployeeID"] != null)
                {
                    int employeeID = (int)card.Sections["PnrPowerAttorney"].Fields["EmployeeID"];
                    SetConfidantsVisibility(employeeID, confidants, confidantNotEmployee);
                }

                // видимость контролов при смене значения "Сотрудник ГК Пионер"
                context.Card.Sections["PnrPowerAttorney"].FieldChanged += (s, e) =>
                {
                    if (e.FieldName == "EmployeeID" && e.FieldValue != null)
                    {
                        SetConfidantsVisibility((int)e.FieldValue, confidants, confidantNotEmployee);
                    }
                };
            }

            return base.Initialized(context);
        }
    }
}
